private/bufpkg/bufmodule/bufmoduleapi/uploader.go

```diff
@@ -494,3 +494,38 @@ func remoteDepToV1Beta1ProtoUploadRequestDepRef(
 		Registry: remoteDep.ModuleFullName().Registry(),
 	}, nil
 }
+
+// In UploadRequest, only dependency commit IDs are included (as opposed to module + commit ID), this means
+// when the BSR fails to resolve the commit ID, it can only return an error reporting this ID is not found,
+// without saying which module the commit ID belongs to.
+//
+// asDependencyNotFoundError returns an error reporting moduleFullName:CommitID not found if err
+// is an error reporting the commitID was not found. Otherwise, it returns the err as is.
+func asDependencyNotFoundError(err error, remoteDeps []bufmodule.RemoteDep) error {
+	if connect.CodeOf(err) != connect.CodeInvalidArgument {
```

Comment:
Leaving this PR as draft because I'm not sure if the server should return `CodeInvalidArgument` when one of the dependencies is not found.